src/engine/guidance/post_processing.cpp

```diff
@@ -812,6 +816,11 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
 //
 bool isStaggeredIntersection(const RouteStep &previous, const RouteStep &current)
 {
+    //don't touch roundabouts
+    if (entersRoundabout(previous.maneuver.instruction) ||
+        entersRoundabout(current.maneuver.instruction))
+        return false;
+
```

Comment:
I think we should completely disable the Staggered Intersection logic within roundabouts (or what about mode changes, or road classes for which it makes no sense, .. ?) - it really only targets the very short zig-zags in rural areas as described in https://github.com/Project-OSRM/osrm-backend/pull/2738. What's your take on this? (Not blocking this PR, though)